ballerina/engine.bal

```diff
@@ -21,11 +21,12 @@ import graphql.parser;
 isolated class Engine {
     private final readonly & __Schema schema;
     private final int? maxQueryDepth;
-    private final readonly & (readonly & Interceptor)[] interceptors;
+    private final readonly & ((readonly & Interceptor)|(readonly & Interceptor)[]) interceptors;
```

Comment:
Can't we define a new type for this? Looks repeating multiple times.

```ballerina
type ReadonlyInterceptor readonly & Interceptor;
```

We can even make this the default type since we do not need non-readonly interceptors in the first place.

Reply:
At the time we added the interceptor support, there was this https://github.com/ballerina-platform/ballerina-lang/issues/36506 lang bug. Since it was fixed now, we can make this as default type. fixed via https://github.com/ballerina-platform/module-ballerina-graphql/pull/1245/commits/0e79ac5618bfd988b2175f4d6fe6e96da485e4af